sdk/src/test/scala/com/horizen/api/http/client/SecureEnclaveApiClientTest.scala

```diff
@@ -32,27 +30,99 @@ class SecureEnclaveApiClientTest extends AnyWordSpec with Matchers with MockitoS
 
   implicit val system: ActorSystem = ActorSystem("test-api-client")
   implicit val timeout: Timeout = Timeout(1.second)
+  private val keySerializer: SchnorrPropositionSerializer = SchnorrPropositionSerializer.getSerializer
   private val signatureSerializer: SchnorrSignatureSerializer = SchnorrSignatureSerializer.getSerializer
-
+  private val mapper = new ObjectMapper()
 
   "Secure Enclave Api should " should {
 
-    "return empty future if it is disabled" in {
+    "tell if it is enabled" in {
       val (apiClient, serverMock) = prepareApiClient(false)
-      val key = generateKey().publicImage()
-      val result = Await.result(
-        apiClient ? SignWithEnclave("test".getBytes, (key, 1)), 1.second
-      ).asInstanceOf[Option[CertificateSignatureInfo]]
+      assert(false.equals(apiClient.isEnabled))
+      verifyNoInteractions(serverMock)
+
+      val (apiClientEnabled, mockEnabled) = prepareApiClient()
+      assert(true.equals(apiClientEnabled.isEnabled))
+      verifyNoInteractions(mockEnabled)
+    }
+
+
+    "return empty future in case of error for listPublicKeys" in {
+      val (apiClient, serverMock) = prepareApiClient()
+      val response = mapper
+        .createObjectNode()
+        .put("error", "Do Androids Dream of Electric Sheep?")
+        .toString
+
+      when(serverMock.singleRequest(any(),any(),any(),any()))
+        .thenReturn(Future.successful(
+          HttpResponse(status = StatusCodes.OK, entity = response)
+        ))
+
+      val result = Await.result(apiClient.listPublicKeys(), 1.second)
 
       assert(result.isEmpty)
-      verifyNoInteractions(serverMock)
+      verify(serverMock).singleRequest(any(), any(), any(), any())
+    }
+
+    "return list of keys for listPublicKeys" in {
+      val (apiClient, serverMock) = prepareApiClient()
+      val key1 = generateKey().publicImage()
+      val key2 = generateKey().publicImage()
+      val key3 = generateKey().publicImage()
+      val key1Json = mapper.createObjectNode()
+        .put("publicKey", BytesUtils.toHexString(keySerializer.toBytes(key1)))
+        .put("type", "schnorr")
+      val key2Json = mapper.createObjectNode()
+        .put("publicKey", BytesUtils.toHexString(keySerializer.toBytes(key2)))
+        .put("type", "schnorr")
+      val key3Json = mapper.createObjectNode()
+        .put("publicKey", BytesUtils.toHexString(keySerializer.toBytes(key3)))
+        .put("type", "schnorr")
+
+      val response = mapper
+          .createObjectNode()
+          .set("keys", mapper.createArrayNode().add(key1Json).add(key2Json).add(key3Json))
+          .toString
+
+      when(serverMock.singleRequest(any(),any(),any(),any()))
+        .thenReturn(Future.successful(
+          HttpResponse(status = StatusCodes.OK, entity = response)
+        ))
+
+      val result = Await.result(apiClient.listPublicKeys(), 1.second)
+
+      assert(result.size.equals(3))
```

Comment:
use `assertEquals` and add message to explain the failure.
Same in other places

Reply:
changed to scala style matchers, they provide a human readable output describing what went wrong